cocotb/triggers.py

```diff
@@ -456,13 +456,20 @@ def __repr__(self):
 class Lock(object):
     """Lock primitive (not re-entrant).
 
-    This should be used as::
+    This can be used as::
 
-        yield lock.acquire()
+        await lock.acquire()
         try:
             # do some stuff
         finally:
             lock.release()
+
+    .. versionchanged:: 1.4
+        The lock can be used as an asynchronous context manager in an
+        :keyword:`async with` statement::
+
+            async with lock:
+                # do some stuff
```

Comment:
Not rendering correctly: https://cocotb--1532.org.readthedocs.build/en/1532/triggers.html#cocotb.triggers.Lock

Perhaps

```suggestion

        The lock can be used as an asynchronous context manager in an
        :keyword:`async with` statement::

            async with lock:
                # do some stuff
```